flake-module.nix

```
@@ -159,9 +202,22 @@ in
                     hlint;
                 };
                 buildTools = lib.attrValues (defaultBuildTools hp // cfg.buildTools hp);
-                package = cfg.modifier (hp.callCabal2nixWithOptions cfg.name cfg.root "" { });
+                package =
+                  if cfg.topLevelPackages.enableMultiPkgs then
+                    hp.${cfg.topLevelPackages.defaultPkg}
+                  else
+                    cfg.modifier (hp.callCabal2nixWithOptions cfg.name cfg.root "" { });
                 devShell = with pkgs.haskell.lib;
-                  (addBuildTools package buildTools).envFunc { withHoogle = true; };
+                  (addBuildTools package buildTools).envFunc (
+                    {
+                      withHoogle = true;
+                    } // (
+                      if cfg.topLevelPackages.enableMultiPkgs then {
+                        packages = pkgs: map (pkg: pkgs.${pkg}) cfg.topLevelPackages.dirs;
+                        buildInputs = [ hp.ghcid ];
```

```suggestion
                        nativeBuildInputs = [ hp.ghcid ];
```

This will provide shell completions and cross compilation support.